test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[NotFoundException] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("not found")
 
-    "result in a BadRequest exception" when {
-      "when a BadRequest http response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(BadRequestHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[BadRequestException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"bad request\""
-      }
     }
 
-    "result in a NotFound exception" when {
-      "when a NotFound http response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(NotFoundHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[NotFoundException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"not found\""
-      }
+    "should return a InternalServerError when INTERNAL_SERVER_ERROR response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(INTERNAL_SERVER_ERROR).withBody("internal server error")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[Upstream5xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("internal server error")
+
     }
 
-    "result in a InternalServerError exception" when {
-      "when a InternalServerError http response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any()))
-          .thenReturn(Future.successful(InternalServerErrorHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[InternalServerException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"internal server error\""
-      }
+    "should return a BadRequestException when BAD_REQUEST response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(BAD_REQUEST).withBody("bad request")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[BadRequestException] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("bad request")
+
     }
 
-    "result in a Locked exception" when {
-      "when a Locked response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(LockedHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[LockedException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"locked\""
-      }
+    "should return a LockedException when LOCKED response" in {
```

Comment:
this is throwing a `Upstream4xxResponse` not a `LockedException`

Reply:
Changed test name to Upstream4xxResponse